core/vdbe/mod.rs

```diff
@@ -316,6 +340,11 @@ impl VTabOpaqueCursor {
     }
 }
 
+pub enum GeneralCursor<'a> {
```

Comment:
tbh both of these are BTree cursors, one of them just uses an in memory BTreeMap/BTreeSet and is not persistent. wonder if we should call this

```
pub enum GeneralBTreeCursor {
  Persistent(&'a mut BTreeCursor),
  Ephemeral(&'a mut EphemeralCursor)
}
```

also, both the persistent and ephemeral cursors share a lot of the same methods and now there's a lot of pattern matching duplication in various opcodes, so wondering if it would make sense to just do it like this:

```
impl <'a> GeneralBTreeCursor<'a> {
  fn rewind(...) {
    match self {
       Self::Persistent(p) => c.rewind(...)),
       Self::Ephemeral(e) => e.rewind(...)),
    }
  }
}
```